agent/test/src/health_test.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:test/test.dart';
+import 'package:cocoon_agent/src/health.dart';
+
+void main() {
+  testRemoveXcodeDerivedData();
+}
+
+void testRemoveXcodeDerivedData() {
+  MemoryFileSystem memfs;
+
+  setUp(() {
+    memfs = MemoryFileSystem();
+    homeEnvVar = () => '/foo';
```

Comment:
I'm not sure this is doing what it looks like it wants to do.  I think the way we would normally achieve this is to override the `Platform.environment` with a `Platform` abstraction.

Reply:
How to override the `Platform.environment` with a `Platform` abstraction? Is there an example?

Reply:
Done.